spec/lib/guardian_extensions_spec.rb

```diff
@@ -267,4 +267,63 @@
       end
     end
   end
+
+  describe "#can_create_channel_message?" do
+    context "when user is staff" do
+      it "returns true if the channel is open" do
+        channel.update!(status: :open)
+        expect(staff_guardian.can_create_channel_message?(channel)).to eq(true)
+      end
+
+      it "returns true if the channel is closed" do
+        channel.update!(status: :closed)
+        expect(staff_guardian.can_create_channel_message?(channel)).to eq(true)
+      end
+
+      it "returns false if the channel is archived" do
+        channel.update!(status: :archived)
+        expect(staff_guardian.can_create_channel_message?(channel)).to eq(false)
+      end
+
+      context "for direct message channels" do
+        it "returns true if enable_personal_messages is false since staff can always send DMs" do
+          channel.update!(status: :open)
+          SiteSetting.enable_personal_messages = false
```

Comment:
Is this site setting useful in the context of this test? I looked at the method which we're testing here and the site setting doesn't seem to be used in the method at all.